docs/DeveloperGuide.md

```diff
@@ -4,10 +4,154 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Parser
+API: [Parser.java]({repoURL}src/main/java/seedu/cafectrl/parser/Parser.java)
 
+![Parser Class Diagram](images/class/Parser.png)
+
+*Figure 1: Parser Class Diagram*
+
+Note that `CafeCtrl` only have access to the interface `ParserUtil` although the run-time type object is `Parser`. With this, we are able to decrease coupling between `CafeCtrl` and `Parser`, allowing for easier maintenance. This also ensures the testability as we could provide mock or stub dependencies during testing, we could isolate the behavior of the class and focus on unit testing without external dependencies. 
+
+![Parser Parsing User Input Sequence Diagram](images/sequence/Parser.png)
+
+*Figure 2: Parser Parsing User Input Sequence Diagram*
+
+When user input a string to `Main`,  it passes the full user input to `Parser` via `parseCommand`. In `parseCommand`,  it finds the matching keyword for different command from the user input, then it calls the respective `prepareCommand` method within `Parser`. `prepareCommand` then generates the corresponding command class and return it to `parseCommand`, which returns the `Command` back to `Main` for execution.
+
+### Storage
+API: [Storage.java]({repoURL}src/main/java/seedu/cafectrl/storage/Storage.java)
+
+![Storage Class Diagram](images/class/Storage.png)
```

Comment:
The storage class diagram does not follow CS2113 standard. You may want to look at the notes. For example, shouldn't have the C logo etc.